### PR TITLE
Bring back developers.diem.com folder and associated CI

### PR DESCRIPTION
**Pull request by CapCap**
## Motivation
Move the developers website from readme.com back into the github

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?
yes

## Test Plan
Manually tested by @CapCap and @nimishamw via netlify build previews

### CODE REVIEW

File: .github/workflows/developer-site-ci.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Diem Developers Website CI
```
Comment (rexhoffman) [inline .github/workflows/developer-site-ci.yml:1]: this should likely live in ci-test.yml now, especially if the site not building on auto should prevent it landing on main.

**Review by rexhoffman — CHANGES_REQUESTED** [review]

**Comment by bmwill** [comment]
Question on this: Would it be preferable to have the website live in its own repo again? I don't remember the reasoning why we moved it into the repo in the first place but it did have some consequences given we wanted to have both the rust documentation and the website hosted on the Github Pages site.

**Comment by davidiw** [comment]
> Question on this: Would it be preferable to have the website live in its own repo again? I don't remember the reasoning why we moved it into the repo in the first place but it did have some consequences given we wanted to have both the rust documentation and the website hosted on the Github Pages site.

Would love to have a doc strategy -- having multiple different places for documentation means we have no strategy, no data, no waterfall, 🤯

**Comment by rexhoffman** [comment]
My view is if it matters, put it in ci-test.yml.    Otherwise no one will see, or care about it.   Make sure to leverage the changes/matches action to only build when relevant changes occur.

**Comment by JoelMarcey** [comment]
It was moved out because the developer.diem.com docs were moved to [Readme](https://readme.com/). Are we moving them back to being served from GitHub and/or Netlify? That would be the only reason why this PR would make sense. Also, the Move docs are part of this too and those are being handled separately.

I have not been in the meetings recently as I have been working on other things, but I would make sure this is what you want to do, maybe at the sync meeting on Monday?

File: developers.diem.com/docs/governance.mdx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+## Diem Improvement Proposals (DIP)
+
+Diem Improvement Proposals (DIPs) offer the Diem developer community a way to participate in advancing the features and functioning of the Diem payment network (DPN), either by proposing a change or engaging in discussion about a proposal. Learn more about how DIPs work [here](https://dip.diem.com/overview). You can also view the current list of [all DIPs]((https://dip.diem.com/all-dips)).
+
+## Technical Steering Committee
+
+The [Technical Steering Committee (TSC)](https://www.diem.com/en-us/about-us/#technical_steering_committee) comprises members who supervise technical design and development of the Diem Blockchain.
+
+Appointed by the Council, the TSC is comprised of representatives from Member organizations who are tasked with advising on the technical design and development of the Diem network.
+
```
Comment (dahliamalkhi) [inline developers.diem.com/docs/governance.mdx:18]: Should we avoid duplicating content here and just leave the reference to the TSC page?

**Review by davidiw — APPROVED** [review]
let's land this sucker
then do a thorough review of the content and design (sending updates where appropriate)
then do the rest of the migration

**Pull request closed without merging**